src/app/screens/signBatchPsbtRequest/index.tsx

```diff
@@ -153,13 +162,21 @@ function SignBatchPsbtRequest() {
         return undefined;
       }
     },
-    [selectedAccount, network.type],
+    [txnContext],
   );
 
-  const parsedPsbts = useMemo(
-    () => payload.psbts.map(handlePsbtParsing),
-    [handlePsbtParsing, payload.psbts],
-  );
+  useEffect(() => {
+    (async () => {
+      const parsedPsbtsResult = await Promise.all(payload.psbts.map(handlePsbtParsing));
```

Comment:
can we make sure that we process the PSBTs in the same order we receive them here? the final response must match the requested order